src/hardware/adc.rs

```diff
@@ -208,18 +206,26 @@ impl Adc {
 
         let (data, status) = self.adcs.read_data();
 
-        set_cs!(self, current_phy, false);
+        self.rdyn.clear_interrupt_pending_bit();
+
+        set_cs!(self, current_phy, true);
 
         self.current_position = (self.current_position + 1) % Self::SCHEDULE.len();
 
-        let (current_phy, ch) = Self::SCHEDULE[self.current_position];
+        let (current_phy, _) = Self::SCHEDULE[self.current_position];
 
-        set_cs!(self, current_phy, true);
+        set_cs!(self, current_phy, false);
```

Comment:
A boolean here isn't readable as it lacks context (what's "false"?). I'd definitely stick with `PinState`.